### PR TITLE
[FX-5059] Migrate circular button

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-5059]

### Description

We need to migrate Button.Circle to TailwindCSS

### How to test

- [Temploy](https://picasso.toptal.net/fx-5059-migrate-circular-button)

### Development checks

- N/A Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-5059]: https://toptal-core.atlassian.net/browse/FX-5059?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]
tested locally, styles are the same 👍

File: packages/picasso-tailwind/src/index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ module.exports = {
     borderRadius: {
       sm: '4px',
       md: '8px',
+      full: '9999px',
```
Comment (sashuk) [inline packages/picasso-tailwind/src/index.js:26]: This is coming from https://github.com/tailwindlabs/tailwindcss/discussions/3373, correct?

File: packages/base/Button/src/ButtonCircular/styles.ts [hunk 2]
```diff
@@ -1,75 +1,127 @@
...
+    disabled?: boolean
+    focused?: boolean
+    hovered?: boolean
+    active?: boolean
+  }
+): string[] => {
+  const variantClassNames = []
+
+  switch (variant) {
+    case 'primary':
```
Comment (sashuk) [inline packages/base/Button/src/ButtonCircular/styles.ts:41]: There is quite some logic, do you think we could have some automated test for it? Probably generating it via GPT and just checking the actual test will be even faster that creating it manually for all cases

**Review by sashuk — APPROVED** [review]
Tested in temploy for different breakpoints and states, left some minor comments, looks great!

File: packages/picasso-tailwind/src/index.js (hunk 1 above)
Comment (OleksandrNechai) [reply packages/picasso-tailwind/src/index.js:26]: If so, why not comment this? Everyone will be stumbling here in the future.

Comment (dmaklygin) [reply packages/picasso-tailwind/src/index.js:26]: Actually, it comes from the default configuration. We have to put it here to keep the consistent look

File: packages/base/Button/src/ButtonCircular/styles.ts (hunk 2 above)
Comment (dmaklygin) [reply packages/base/Button/src/ButtonCircular/styles.ts:41]: I think, we should rely on visual tests that we have. Snapshot testing would be odd here, wdyt?

Comment (sashuk) [reply packages/base/Button/src/ButtonCircular/styles.ts:41]: @dmaklygin If we cover all states in visual tests then should be fine 👍

**Pull request merged**